rust/foxglove/src/mcap_writer.rs

```diff
@@ -8,22 +8,32 @@ use std::{fmt::Debug, io::Write};
 
 use crate::library_version::get_library_version;
 use crate::sink_channel_filter::SinkChannelFilterFn;
-use crate::{ChannelDescriptor, Context, FoxgloveError, Sink, SinkChannelFilter};
+use crate::{ChannelDescriptor, Context, FoxgloveError, Sink, SinkChannelFilter, SinkId};
 
 /// Compression options for content in an MCAP file
 pub use mcap::Compression as McapCompression;
 /// Options for use with an [`McapWriter`][crate::McapWriter].
 pub use mcap::WriteOptions as McapWriteOptions;
 
+#[cfg(feature = "live_visualization")]
+mod async_mcap_sink;
 mod mcap_sink;
+
+#[cfg(feature = "live_visualization")]
+use async_mcap_sink::AsyncMcapSink;
 use mcap_sink::McapSink;
 
 /// An MCAP writer for logging events.
 ///
-/// ### Buffering
+/// ### Methods
+///
+/// - [`create`](McapWriter::create) - Synchronous. Writes directly, may block on disk I/O.
```

Comment:
_Technically_ this can write to anything that a Writer can write to, not just disk. It's worth noting for your context that `create_new_buffered_file` is just a convenience wrapper around making a BufWriter and passing it to `create`